cookbooks/bcpc-hadoop/templates/default/hb_hbase-site.xml.erb

```diff
@@ -166,4 +156,138 @@
     <value>org.apache.hadoop.hbase.ipc.SecureRpcEngine</value>    
   </property>   
   <% end %>
+
+  <property>
+    <name>dfs.client.read.shortcircuit</name>
+    <value><%= node["bcpc"]["hadoop"]["hbase"]["shortcircuit"]["read"]%></value>
+  </property>
+
+  <% if node["bcpc"]["hadoop"]["hbase"]["shortcircuit"]["read"] == true then %>
+  <property>
+    <name>dfs.domain.socket.path</name>
+    <value>/var/run/hadoop-hdfs/dn._PORT</value>
+  </property>
+
+  <property>
+    <name>dfs.client.read.shortcircuit.buffer.size</name>
+    <value>131072</value>
+  </property>
+  <% end %>
+
+  <property>
+    <name>hbase.regionserver.handler.count</name>
+    <value>128</value>
+  </property>
+
+  <!-- Log responses which take longer than 250ms -->
+  <property>
+    <name>hbase.ipc.warn.response.time</name>
+    <value><%= node["bcpc"]["hadoop"]["hbase"]["ipc"]["warn"]["responsetime"]%></value>
+  </property>
+
+  <!-- Log responses which are larger than 1MB -->
+  <property>
+    <name>hbase.ipc.warn.response.size</name>
+    <value>1048576</value>
+  </property>
+
+  <property>
+    <name>hbase.ipc.server.tcpnodelay</name>
+    <value>true</value>
+  </property>
+
+  <% 
+  bucketcache_size = (node["bcpc"]["hadoop"]["hbase_rs"]["xmx"]["size"] * node["bcpc"]["hadoop"]["hbase"]["blockcache"]["size"] + node["bcpc"]["hadoop"]["hbase_rs"]["mx_dir_mem"]["size"]  -  node["bcpc"]["hadoop"]["hbase_rs"]["hdfs_dir_mem"]["size"]).floor 
+  bucketcache_combinedcache_percent = bucketcache_size.to_f/(node["bcpc"]["hadoop"]["hbase_rs"]["xmx"]["size"] + node["bcpc"]["hadoop"]["hbase_rs"]["mx_dir_mem"]["size"] -  node["bcpc"]["hadoop"]["hbase_rs"]["hdfs_dir_mem"]["size"])
+  %>
```

Comment:
@cbaenziger. The computation was placed closer to where is it used so that it is easier for code readability. Will move it to the top of the template.